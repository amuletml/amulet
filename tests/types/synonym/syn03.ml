type two_of 'a <- 'a * 'a

let foo : list (two_of int) = [(1, 1)]
let bar (x : list (int * int)) = ()

let _ = bar foo
