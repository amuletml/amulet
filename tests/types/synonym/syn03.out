two_of : Infer{'a : type}. 'a -> 'a
foo : list (int * int)
bar : list (int * int) -> unit
